core-plugins/src/main/java/io/cdap/plugin/transform/LogParserTransform.java

```diff
@@ -196,7 +208,7 @@ private String getLog(StructuredRecord input) {
     Schema.Type inputType = inputSchema.getType();
 
     try {
-      validateInputSchemaType(inputType);
+      validateInputSchemaType(inputSchema, config.inputName, getContext().getFailureCollector());
```

Comment:
check in line 200 should be done through validation

Reply:
For that function, should we still be returning null on failure or should we be changing the getLog function to use exceptions for error handling?

Reply:
Oh also how can we verify that the transform context is not null when retrieving the failure collector?

Reply:
Created a new function that accepts an overload. Since this is only used in transform, we will keep the existing error logging structure.